content-services/latest/config/mtls.md

```diff
@@ -0,0 +1,283 @@
+|keysize|2048/4096|Specifies the RSA key length. The default value is `2048`.|
+|keystoretype|PKCS12/JKS/JCEKS|Sets the type of the keystore (containing private keys). The default value is `JCEKS`.|
+|keystorepass|Any string between 6 and 1023 characters|Specifies the password for the keystore. A prompt will be shown for the default value.|
+|notruststore|N/A|Flag to turn off generating of the truststore and needs to be defined. The default value is N/A.|
+|truststoretype|JKS/JCEKS|Sets the type of the truststore (containing public keys). The default value is `JCEKS`.|
+|truststorepass|Any string between 6 and 1023 characters|Specifies the password for the truststore. A prompt will be shown for the default value.|
+|certdname|String|Sets the Distinguished Name of the CA certificate, starting with a forward-slash. The default value is `/C=GB/ST=UK/L=Maidenhead/O=Alfresco Software Ltd./OU=Unknown/CN=Custom Service`.|
+|servername|String|DNS Names for Service. Multiple values can be provided, split by “,”. For example: `localhost`,`additional`. In Windows, these have to be enclosed in double quotes. The default value is `localhost`.|
+|alfrescoformat|classic/current|The default format for certificates is `current` for IE SS 2.0.0+ and `classic` for previous versions. The main difference is that classic format generates text files containing keystore/truststore password, private key/public key alias, and in case of keystore, also the private key password (which currently has to be the same as the keystore). The default value is `current`.|
+
+> **Note:** If you plan to use custom DNames in your certificates, you must use double quotes around the values. For example:
+
+```bash
+$ ./run.sh -cacertdname  
+"/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Windows Alfresco CA"
+-repocertdname "/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Repo"
+-solrcertdname "/C=GB/ST=UK/L=Maidenhead/O=Alfresco/OU=Unknown/CN=Solr"
+```
+
+It is recommended that you set your own passwords when generating certificates. For example:
+
+(For Linux)
+
+```bash
+./run.sh -keystorepass "password" -truststorepass "password"
+```
+
+(For Windows)
+
+```bash
+run.cmd -keystorepass "password" -truststorepass "password"
+```
+
+### Additional script examples
+
+Find more sample scripts in [alfresco-ssl-generator](https://github.com/Alfresco/alfresco-ssl-generator) repository [for Linux](https://github.com/Alfresco/alfresco-ssl-generator/tree/master/ssl-tool/samples) and [for Windows](https://github.com/Alfresco/alfresco-ssl-generator/tree/master/ssl-tool-win/samples).
+
+> **Note:** In Windows:
+>    * To run sample scripts you’ll need to move them 1 directory upwards
+>    * Add keytool to PATH system variable
+>    * Set 2 environment variables: `SET OPENSSL_CONF=<path_to_project>\alfresco-ssl-generator\ssl-tool-win\openssl.cnf` and `SET RANDFILE=.rnd`
+
+## Keystore directory structure
+
+The `keystores` directory contains the following structure and files:

```

Comment:
I'd add a sentence here like
"For old (run.sh/run.cmd) script, the keystores directory contains the following structure and files.
For new scripts: run_encryption, run_additional; the folders within "keystores" folder can be specified through "subfoldername" parameter. The keystore and truststore name will use "servicename" parameter value to create files.

Reply:
Or add the "For new scripts..." at the bottom of "Keystore directory structure"

Reply:
I reversed the order to expalin the old versus new keystore structure. Please let me know if it needs correction.

Reply:
I've updated the documentation.